src/core/analysis/mcmc/SteppingStoneSampler.cpp

```diff
@@ -73,8 +78,102 @@ double SteppingStoneSampler::marginalLikelihood( void ) const
+ * does not work well. We collect these warnings and present them to the user after the calculation has been completed.
+ *
+ * @return The approximate standard error
+ */
+double SteppingStoneSampler::standardError( void ) const
+{
+    double vmlnl = 0.0;
+    
+    if ( process_active == true )
+    {
+        std::stringstream warnings;
+        size_t warning_counter = 0;
+        
+        for (size_t i = 1; i < powers.size(); ++i)
+        {
+            
+            size_t samplesPerPath = likelihoodSamples[i].size();
+            
+            // compute the maximum of the log likelihood values
+            double max = likelihoodSamples[i][0];
+            for (size_t j = 1; j < samplesPerPath; ++j)
+            {
+                if (max < likelihoodSamples[i][j])
+                {
+                    max = likelihoodSamples[i][j];
+                }
+            }
+            
+            /* For numerical stability, we first subtract the maximum sampled log likelihood from the rest.
+             * We then multiply the difference by the difference of successive powers, and exponentiate the result.
+             * This is equivalent to dividing by the largest sampled likelihood term, and raising the result to
+             * the difference of successive powers (Xie et al. 2011: 153).
+             */
+            std::vector<double> Ls(samplesPerPath);
+            for (size_t j = 0; j < samplesPerPath; ++j)
+            {
+                Ls[j] = exp( (likelihoodSamples[i][j] - max)*(powers[i - 1] - powers[i]) );
+            }
+            
+            // get the variance and the mean of the resulting values
+            double mean_Ls = 0.0;
+            double var_Ls = 0.0;
+            
+            for (size_t j = 0; j < samplesPerPath; ++j)
+            {
+                mean_Ls += Ls[j];
+            }
+            mean_Ls /= samplesPerPath;
+            
+            for (size_t j = 0; j < samplesPerPath; ++j)
+            {
+                var_Ls += (Ls[j] - mean_Ls)*(Ls[j] - mean_Ls);
+            }
+            var_Ls /= (int(samplesPerPath) - 1);
+            
+            // get the effective sample size and divide the variance by it
+            double ess = getESS(Ls);
+            double vzr = var_Ls / ess;
+            
+            if (vzr / (mean_Ls*mean_Ls) > 0.1)
+            {
+                warning_counter++;
+                if ( warning_counter == 1 )
+                {
+                    warnings << "Warning: Standard error approximation unreliable as var(r_k)/r_k^2 > 0.1 for the following powers:\n" << powers[i];
+                }
+                else
+                {
+                    warnings << ", " << powers[i];
+                }
+            }
+            
+            vmlnl += vzr / (mean_Ls*mean_Ls);
+            
+        }
+        
+        if ( not warnings.str().empty() )
```

Comment:
nitpick but you can use warning_counter here

Reply:
Done